staging/src/k8s.io/apiserver/pkg/util/flowcontrol/request/mutating_work_estimator.go

```diff
@@ -69,24 +78,56 @@ func (e *mutatingWorkEstimator) estimate(r *http.Request) WorkEstimate {
 	// we will work on tuning the algorithm later. As a starting point we
 	// we simply assume that processing 1 event takes 1/Nth of a seat for
 	// M milliseconds and processing different events is infinitely parallelizable.
-	// We simply record the appropriate values here and rely on potential
-	// reshaping of the request if the concurrency limit for a given priority
-	// level will not allow to run request with that many seats.
-	//
+	// We simply estimate the total amount of final work for this request based
+	// on the above and rely on potential reshaping of the request if the
+	// concurrency limit for a given priority level will not allow to run
+	// request with that many seats.
```

Comment:
This is no longer true.

Reply:
Which one - in my opinion it is true:
- we estimate the total amount of final work based on the this assumption
- we still rely on potential reshaping if concurrency limit is smaller then final seats.

Reply:
First, the preceding unchanged sentence is a bit off base.  It says "we assume ... is infinitely parallelizable".  Was that intended to say that the actual work is all launched at once?  Or that we can freely reshape that work as much as we like?

I think a more accurate statement, starting with a revised version of the preceding thus-far-unchanged sentence, would be something like the following.

> // As a starting point we assume that the actual work associated with the watchers happens
> // in many waiting goroutines that are all resumed at once,
> // each such goroutine taking 1/Nth of a seat for M milliseconds.
> // We allow the accounting in APF of that work to be reshaped into
> // another rectangle of equal area, for practical reasons.

Reply:
Slightly changed the version (as the multiple gorotuines part is not an assumption - it's a fact).